### PR TITLE
Added option for smoothing images in gammapy.maps

### PR DESCRIPTION
**Pull request by AtreyeeS**
Changed the plot() to plot images on astropy.visualization.wcsaxes
The image can be smoothed during the plotting.

### CODE REVIEW

File: gammapy/maps/wcsnd.py [hunk 1]
```diff
@@ -460,51 +460,150 @@ def downsample(self, factor, preserve_counts=True):
...
         import matplotlib.pyplot as plt
-        import matplotlib.colors as colors
-
-        if ax is None:
+        from astropy.visualization import simple_norm
+        from scipy.ndimage import gaussian_filter
+        from scipy.signal import convolve
+        from scipy.stats import gmean
+        from astropy.convolution import Tophat2DKernel
+        from astropy.coordinates import Angle
```
Comment (adonath) [inline gammapy/maps/wcsnd.py]: Please move all imports to the top, except for the optional dependencies such as `scipy`.

File: gammapy/maps/wcsnd.py [hunk 2]
```diff
@@ -460,51 +460,150 @@ def downsample(self, factor, preserve_counts=True):
...
+            ax.coords['dec'].set_axislabel('Declination')
+        except AttributeError:
+            log.info("Can't set coordinate axes. No WCS information available.")
+
+        # without this the axis limits are changed when calling scatter
+        ax.autoscale(enable=False)
+        return fig, ax, cbar
+
+
+    def peek_at_pos(pos,axis,idx=None):
```
Comment (adonath) [inline gammapy/maps/wcsnd.py]: Please remove this function, that is not implemented yet. You can add it any time in a second pull request.

File: gammapy/maps/wcsnd.py [hunk 3]
```diff
@@ -460,51 +460,150 @@ def downsample(self, factor, preserve_counts=True):
...
+        TODO: let idx work with slicing
+
+        fig : `~matplotlib.figure.Figure`, optional
+            Figure
+        stretch : str, optional
+            Scaling for image ('linear', 'sqrt', 'log').
+            Similar to normalize and stretch functions in ds9.
+            See http://docs.astropy.org/en/stable/visualization/normalization.html
+
+        smooth: ~string : either 'gauss' or 'disk'
```
Comment (adonath) [inline gammapy/maps/wcsnd.py]: Please change the documentation of the smoothing parameters to the same format  as the other parameters of the `.plot()` function.

Comment (cdeil) [reply gammapy/maps/wcsnd.py]: Suggest to use this format for multiple-choice option parameters:
```
When a parameter can only assume one of a fixed set of values, those values can be listed in braces, with the default appearing first:

order : {'C', 'F', 'A'}
    Description of `order`.
```
https://numpydoc.readthedocs.io/en/latest/format.html

**Comment by AtreyeeS** [comment]
Hello, 

I was having some problems rebasing my master, so I made a new branch (maps1) and made the changes. Should I delete this pull request and make a new one?

regards,

**Comment by adonath** [comment]
@AtreyeeS Yes, if it's easier just open a new PR from your new branch and we'll close this one.

**Comment by adonath** [comment]
Superseded by #1431.

**Pull request closed without merging**